kvdb-rocksdb/src/lib.rs

```diff
@@ -443,15 +443,22 @@ impl Database {
 							stats_total_bytes += key.len();
 							batch.delete_cf(cf, &key).map_err(other_io_err)?
 						}
-						DBOp::DeletePrefix { col: _, prefix } => {
+						DBOp::DeletePrefix { col, prefix } => {
 							if !prefix.is_empty() {
-								let end_range = kvdb::end_prefix(&prefix[..]);
-								batch.delete_range_cf(cf, &prefix[..], &end_range[..]).map_err(other_io_err)?;
+								if let Some(end_range) = kvdb::end_prefix(&prefix[..]) {
+									batch.delete_range_cf(cf, &prefix[..], &end_range[..]).map_err(other_io_err)?;
+								} else {
+									for (key, _) in self.iter_with_prefix(col, &prefix[..]) {
+										batch.delete_cf(cf, &key[..]).map_err(other_io_err)?;
+									}
+								}
 							} else {
 								// Deletes all values in the column.
-								let end_range = &[u8::max_value()];
-								batch.delete_range_cf(cf, &[][..], &end_range[..]).map_err(other_io_err)?;
-								batch.delete_cf(cf, &end_range[..]).map_err(other_io_err)?;
+								let end_range = [u8::max_value(); 16];
+								batch.delete_range_cf(cf, &prefix[..], &end_range[..]).map_err(other_io_err)?;
+								for (key, _) in self.iter_with_prefix(col, &end_range[..]) {
+									batch.delete_cf(cf, &key[..]).map_err(other_io_err)?;
+								}
```

Comment:
let's extract this into a function/closure and use it above as well, when the prefix is not empty, but is [255, 255, ...]

Reply:
although it would work for `prefix.len() < 16` case